code/datums/mutations/body.dm

```diff
@@ -128,10 +132,71 @@
 	REMOVE_TRAIT(owner, TRAIT_DWARF, GENETIC_MUTATION)
 	owner.visible_message(span_danger("[owner] suddenly grows!"), span_notice("Everything around you seems to shrink.."))
 
+/datum/mutation/human/acromegaly
+	name = "Acromegaly"
+	desc = "A mutation believed to be the cause of acromegaly, or 'being unusually tall'."
+	quality = MINOR_NEGATIVE
+	difficulty = 16
+	instability = NEGATIVE_STABILITY_MODERATE
+	synchronizer_coeff = 1
+	conflicts = list(/datum/mutation/human/dwarfism)
+
+/datum/mutation/human/acromegaly/on_acquiring(mob/living/carbon/human/owner)
+	if(..())
+		return
+	ADD_TRAIT(owner, TRAIT_TOO_TALL, GENETIC_MUTATION)
+	owner.visible_message(span_danger("[owner] suddenly grows tall!"), span_notice("You feel a small strange urge to fight small men with slingshots. Or maybe play some basketball."))
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(head_bonk))
+	owner.regenerate_icons()
+
+/datum/mutation/human/acromegaly/on_losing(mob/living/carbon/human/owner)
+	if(..())
+		return
+	REMOVE_TRAIT(owner, TRAIT_TOO_TALL, GENETIC_MUTATION)
+	owner.visible_message(span_danger("[owner] suddenly grows!"), span_notice("You return to your usual height."))
+	UnregisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(head_bonk))
+	owner.regenerate_icons()
+
+// This is specifically happening because they're not used to their new height and are stumbling around into machinery made for normal humans
+/datum/mutation/human/acromegaly/proc/head_bonk(mob/living/parent)
+	SIGNAL_HANDLER
+	var/turf/airlock_turf = get_turf(parent)
+	var/atom/movable/whacked_by = locate(/obj/machinery/door/airlock) in airlock_turf|| locate(/obj/machinery/door/firedoor) in airlock_turf || locate(/obj/structure/mineral_door) in airlock_turf
```

Comment:
```suggestion
	var/atom/movable/whacked_by = locate(/obj/machinery/door/airlock) in airlock_turf || locate(/obj/machinery/door/firedoor) in airlock_turf || locate(/obj/structure/mineral_door) in airlock_turf
```
Also you include mineral doors but don't include non-airlock doors (windoors, shutters / blast doors, puzzle doors). I assume intentional, just mentioning it if not

Reply:
meh those are either overkill or don't fit imo